kotlin-insight-client/kotlin-insight-client-sdk/src/main/kotlin/com/linkedplanet/kotlininsightclient/sdk/SdkInsightAttachmentOperator.kt

```diff
@@ -57,48 +61,56 @@ object SdkInsightAttachmentOperator : InsightAttachmentOperator {
                 .map(::beanToInsightAttachment)
         }
 
-    override suspend fun downloadAttachment(url: String): Either<InsightClientError, ByteArray> =
+    override suspend fun downloadAttachment(url: String): Either<InsightClientError, InputStream> =
         catchAsInsightClientError {
             val attachmentId = attachmentUrlResolver.parseAttachmentIdFromPathInformation(url)
             val attachmentBean = objectFacade.loadAttachmentBeanById(attachmentId)
-            val inputStream =
-                fileManager.getObjectAttachmentContent(attachmentBean.objectId, attachmentBean.nameInFileSystem)
-            inputStream.readBytes()
+            fileManager.getObjectAttachmentContent(attachmentBean.objectId, attachmentBean.nameInFileSystem)
         }
 
-    override suspend fun downloadAttachmentZip(objectId: InsightObjectId): Either<InsightClientError, ByteArray> =
+    override suspend fun downloadAttachmentZip(objectId: InsightObjectId): Either<InsightClientError, InputStream> =
         either {
-            val attachments = getAttachments(objectId).bind()
-            val fileMap: List<Pair<String, ByteArray>> = attachments.map { attachment ->
-                val attachmentContent = downloadAttachment(attachment.url).bind()
-                attachment.filename to attachmentContent
+            val fileMap = allAttachmentStreamsForInsightObject(objectId).bind()
+            zipInputStreamForMultipleInputStreams(fileMap).bind()
+        }
+
+    private fun allAttachmentStreamsForInsightObject(objectId: InsightObjectId) =
+        catchAsInsightClientError {
+            val attachmentBeans = objectFacade.findAttachmentBeans(objectId.value)
+            attachmentBeans.map { bean ->
+                val attachmentContent = fileManager.getObjectAttachmentContent(bean.objectId, bean.nameInFileSystem)
+                bean.filename to attachmentContent
             }
-            ByteArrayOutputStream().use { byteOutputStream ->
-                ZipOutputStream(byteOutputStream).use { zip ->
-                    fileMap.forEach {
-                        val zipEntry1 = ZipEntry(it.first)
-                        zip.putNextEntry(zipEntry1)
-                        zip.write(it.second)
-                        zip.closeEntry()
+        }
+
+    private fun zipInputStreamForMultipleInputStreams(
```

Comment:
Maybe move this method as util into the api module?

Reply:
hat in der API nix verloren. Da nehme ich lieber die Code Duplikation mit.
Evtl. brauchen wir doch noch ein gemeinsames utiltiy Projekt :(